### DISCUSSION TITLE
Using variables in packageRules

### DISCUSSION THREAD

**Discussion by btxbtx** [renovatebot/renovate#20497] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.133.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I am able to specify an environment variable in my package rules, i.e.

```js
    {
      groupName: "someGroup",
      matchManagers: ["helm-values"],
      matchPaths: ["path/**/to/some/dir/$ENVIRONMENT_VARIABLE"],
    },
```
with ENVIRONMENT_VARIABLE="toast"
and renovate will then match `"path/**/to/some/dir/toast"`

The end goal here is for renovate to run on one file at a time, only when invoked manually, and the filename should be passed to renovate

### What you tried so far.

I've tried using `.process.env.<env>` and `$ENVIRONMENT_VARIABLE` in the matchPaths, but it matches strings literally.

### Relevant debug logs

No relevant logs.

**Comment by rarkins** [comment, marked as answer]
You would need to do this in bot (global) config, using `config.js` with string interpolation

**Reply by secustor** [reply]
:thinking_face: maybe we could allow to template matcher fields.

**Reply by rarkins** [reply]
Yes similar to `{{secrets.A}}` but not secret?

**Reply by secustor** [reply]
I think we are not templating package rules at all.

**Reply by btxbtx** [reply]
Sorry, I have no experience with `.js`, but I am doing all this config in `config.js`.

Here is a watered down version of the file, pasting just to show the package rules.

```js
module.exports = {
  onboardingConfig: {
    globalExtends: ["config:base", ":pinVersions", ":rebaseStalePrs"],
  },

  enabledManagers: ["helm-values"],

  packageRules: [
    {
      groupName: "some-group",
      matchManagers: ["helm-values"],
      matchPaths: ["my/path/to/values/files./${NAME}"],
    },
  ],
};
```

NAME is an env variable.

If I check the logs, I see `"my/path/to/values/files./${NAME}"` -- meaning the string is still matched literally

**Reply by rarkins** [reply]
Do this:

```
module.exports = {
  onboardingConfig: {
    globalExtends: ["config:base", ":pinVersions", ":rebaseStalePrs"],
  },

  enabledManagers: ["helm-values"],

  packageRules: [
    {
      groupName: "some-group",
      matchManagers: ["helm-values"],
      matchPaths: ["my/path/to/values/files./" + process.env.NAME],
    },
  ],
};
```

**Reply by btxbtx** [reply]
I see, thank you.  Works as described for my use case, closing the issue.
